pkg/commands/loaders/branches.go

```diff
@@ -114,77 +115,84 @@ func (self *BranchLoader) obtainBranches() []*models.Branch {
 
 	trimmedOutput := strings.TrimSpace(output)
 	outputLines := strings.Split(trimmedOutput, "\n")
-	branches := make([]*models.Branch, 0, len(outputLines))
-	for _, line := range outputLines {
+
+	branches := slices.FilterMap(outputLines, func(line string) (bool, *models.Branch) {
```

Comment:
FilterMap is good when you need to obtain some information as part of the filtering step that you also need in the mapping step. It is annoying to have to return two values however. In this case I've factored out a parseBranch method so that that method can just return a branch directly and the two-value stuff is minimised